resources/measures/advanced_rtu_control/measure.rb

```diff
@@ -0,0 +1,335 @@
+    spaces.each do |space|
+      floor_area = space.floorArea
+      sum_floor_area += floor_area
+
+      number_of_people = space.numberOfPeople
+      sum_number_of_people += number_of_people
+
+      volume = space.volume
+      sum_volume += volume
+
+      dsn_oa = space.designSpecificationOutdoorAir
+      next if dsn_oa.empty?
+
+      dsn_oa = dsn_oa.get
+
+      # compute outdoor air rates in case we need them
+      oa_for_people = number_of_people * dsn_oa.outdoorAirFlowperPerson
+      oa_for_floor_area = floor_area * dsn_oa.outdoorAirFlowperFloorArea
+      oa_rate = dsn_oa.outdoorAirFlowRate
+      oa_for_volume = volume * dsn_oa.outdoorAirFlowAirChangesperHour / 3600
+
+      # First check if this space uses the Maximum method and other spaces do not
+      if dsn_oa.outdoorAirMethod == 'Maximum'
+        sum_oa_rate += [oa_for_people, oa_for_floor_area, oa_rate, oa_for_volume].max
+      elsif dsn_oa.outdoorAirMethod == 'Sum'
+        sum_oa_for_people += oa_for_people
+        sum_oa_for_floor_area += oa_for_floor_area
+        sum_oa_rate += oa_rate
+        sum_oa_for_volume += oa_for_volume
+      end
+    end
+
+    tot_oa_flow_rate += sum_oa_for_people
+    tot_oa_flow_rate += sum_oa_for_floor_area
+    tot_oa_flow_rate += sum_oa_rate
+    tot_oa_flow_rate += sum_oa_for_volume
+
+    # Convert to cfm
+    tot_oa_flow_rate_cfm = OpenStudio.convert(tot_oa_flow_rate, 'm^3/s', 'cfm').get
+
+    #runner.registerInfo("For #{thermal_zone.name}, design min OA = #{tot_oa_flow_rate_cfm.round} cfm.")
+
+    return tot_oa_flow_rate
+  end
+
+  # define what happens when the measure is run
+  def run(model, runner, user_arguments)
+    super(model, runner, user_arguments)  # Do **NOT** remove this line
+
+    # use the built-in error checking
+    if !runner.validateUserArguments(arguments(model), user_arguments)
+      return false
+    end
+	#read in arguments
+	add_econo = runner.getBoolArgumentValue('add_econo', user_arguments)
+	add_dcv = runner.getBoolArgumentValue('add_dcv', user_arguments)
+	
+	#Set constants
+	fan_tot_eff = 0.7 #Eplus default, reasonable
+    fan_mot_eff = 0.9 #Eplus default, reasonable 
+	
+	#set airflow design ratios
+	max_flow = 0.9
+	min_flow = 0.4 
+	
+	min_flow_fraction = 0.3 #typ for non-inverter driven motors 
+	
+	
+	#Sizing run 
+	standard = Standard.build('90.1-2013')
+    if standard.model_run_sizing_run(model, "#{Dir.pwd}/advanced_rtu_control") == false
+      runner.registerError('Sizing run for Hardsize model failed, cannot hard-size model.')
+      puts('Sizing run for Hardsize model failed, cannot hard-size model.')
+      puts("directory: #{Dir.pwd}")
+      return false
+    end
+
+    #apply sizing values
+    model.applySizingValues
+	
+
+   	model.getAirLoopHVACs.sort.each do |air_loop_hvac|
+	    if air_loop_hvac_unitary_system?(air_loop_hvac) #applying to unitary systems 
+		   #set control type
+			air_loop_hvac.supplyComponents.each do |component|#more efficient way of doing this? 
```

Comment:
To answer your question: you can use model.airLoopUnitarySystems to get all unitary systems in the model. From there, you can pull the thermal zone, air loop, etc. I am not sure if this is necessarily easier, but it removes the need to loop through supply components.

Reply:
I think this section could use a few more applicability checks: 
1) this should only apply to RTUs, so check to make sure there is outdoor air
2) this should not apply to residential systems, which also use unitary system type. The outdoor air part probably takes care of this, but I also like to skip if "residential" is in the name.
3) We should skip any system that uses evap cooling, since they operate differently.
4) We should skip any DOAS system. You can check if any form of "DOAS" is in the name, as well as OS Standard's DOAS check method.
5) We should skip any multizone system, so check to make sure applicable air loops only have one thermal zone, and that they do not already have VAV damper (some of our VAV systems only get assigned a single thermal zone, so this is how we can get around that).

Reply:
Good points--addressed these.